src/status_im2/contexts/syncing/setup_syncing/view.cljs

```diff
@@ -38,31 +38,35 @@
 
 (defn view
   []
-  (let [profile-color (rf/sub [:profile/customization-color])
-        valid-for-ms  (reagent/atom code-valid-for-ms)
-        code          (reagent/atom nil)
-        delay         (reagent/atom nil)
-        timestamp     (reagent/atom nil)
-        set-code      (fn [connection-string]
-                        (when (sync-utils/valid-connection-string? connection-string)
-                          (reset! timestamp (* 1000 (js/Math.ceil (/ (datetime/timestamp) 1000))))
-                          (reset! delay 1000)
-                          (reset! code connection-string)))
-        clock         (fn []
-                        (if (pos? (- code-valid-for-ms
-                                     (- (* 1000 (js/Math.ceil (/ (datetime/timestamp) 1000)))
-                                        @timestamp)))
-                          (swap! valid-for-ms (fn [_]
-                                                (- code-valid-for-ms
-                                                   (- (* 1000
-                                                         (js/Math.ceil (/ (datetime/timestamp) 1000)))
-                                                      @timestamp))))
-                          (reset! delay nil)))
+  (let [{:keys [key-uid customization-color]} (rf/sub [:profile/multiaccount])
+        valid-for-ms (reagent/atom code-valid-for-ms)
+        code (reagent/atom nil)
+        delay (reagent/atom nil)
+        timestamp (reagent/atom nil)
+        set-code (fn [connection-string]
+                   (when (sync-utils/valid-connection-string? connection-string)
+                     (reset! timestamp (* 1000 (js/Math.ceil (/ (datetime/timestamp) 1000))))
+                     (reset! delay 1000)
+                     (reset! code connection-string)))
+        clock (fn []
+                (if (pos? (- code-valid-for-ms
+                             (- (* 1000 (js/Math.ceil (/ (datetime/timestamp) 1000)))
```

Comment:
This repeated rounding mechanism in the code should ideally be in `utils.number`. You can try the `utils.number/naive-round` and see if it helps. The rounding mechanism is different, but you can round up to a certain number of decimal places too. The code would look like `(utils.number/naive-round (datetime/timestamp) 3)`

Reply:
I will take a look, but slightly outside the scope of this pr as this code was there from before, albeit I am the one who added it 🙃